skll/learner.py

```diff
@@ -72,24 +81,42 @@
                         [{'learning_rate': [0.01, 0.1, 1.0, 10.0, 100.0]}],
                         AdaBoostRegressor:
                         [{'learning_rate': [0.01, 0.1, 1.0, 10.0, 100.0]}],
+                        BayesianRidge:
+                        [{'alpha_1': [1e-6, 1e-4, 1e-2, 1, 10],
+                          'alpha_2': [1e-6, 1e-4, 1e-2, 1, 10],
+                          'lambda_1': [1e-6, 1e-4, 1e-2, 1, 10],
+                          'lambda_2': [1e-6, 1e-4, 1e-2, 1, 10]}],
                         DecisionTreeClassifier:
                         [{'max_features': ["auto", None]}],
                         DecisionTreeRegressor:
                         [{'max_features': ["auto", None]}],
                         DummyClassifier:
                         [{}],
+                        DummyRegressor:
+                        [{}],
                         ElasticNet:
                         [{'alpha': [0.01, 0.1, 1.0, 10.0, 100.0]}],
                         GradientBoostingClassifier:
                         [{'max_depth': [1, 3, 5]}],
                         GradientBoostingRegressor:
                         [{'max_depth': [1, 3, 5]}],
+                        HuberRegressor:
+                        [{'epsilon': [1.05, 1.35, 1.5, 2.0, 2.5, 5.0],
+                          'alpha': [1e-4, 1e-3, 1e-3, 1e-1, 1, 10, 100, 1000]}],
                         KNeighborsClassifier:
                         [{'n_neighbors': [1, 5, 10, 100],
                           'weights': ['uniform', 'distance']}],
                         KNeighborsRegressor:
                         [{'n_neighbors': [1, 5, 10, 100],
                           'weights': ['uniform', 'distance']}],
+                        MLPClassifier:
+                        [{}],
```

Comment:
I know searching over `hidden_layer_sizes` takes too long, but what about the other parameters like `activation`, `solver`, `learning_rate`, etc?

Reply:
I didn't include the other parameters either because:

(a) Many of the hyper-parameters are conditionally dependent on each other. For example, `learning_rate` is _only_ used when `solver` = `sgd`,  `learning_rate_init` is _only_ used when `solver` = `sgd` or `adam`, and `early_stopping` is _only_ used when `solver` = `sgd`. So, it's very different from other learners in that sense and there isn't an easy way to put together a `product` over all the parameters in the grid that will all be valid together.

(b) Many of the hyper-parameters shouldn't just be blindly iterated over. For example, the `lbfgs` is not great when you have a lot of data but when you have a small amount of data, it's better but still quite slow. Whether or not to use early stopping interacts with whether you have sufficient data in the first place. 

IMO, these decisions about which sets of hyper-parameters should be searched together depends on the experiment and the data and are best left to the user. However, if you have a suggestion for a default parameter grid that makes sense (without a lot of extra logic to deal with conditional dependence) then I am happy to try it out.

Reply:
Hmm, an option could be to have `sgd` solver and a single hidden layer be the default for any MLP and then have a grid iterate over `activation`, `learning_rate`, and `learning_rate_init`. In that case, if you want to do something different, you will have to change the solver via `fixed_parameters` and then also specify your own set of parameters to search via `param_grids`. Thoughts? @Lguyogiro @aoifecahill @mulhod ?